networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/reputation/ReputationManager.java

```diff
@@ -13,156 +13,44 @@
-    return peerReputations.get(peerAddress.getId(), key -> new Reputation());
-  }
-
-  private static class Reputation {
-    private static final int DEFAULT_SCORE = 0;
-    private static final EnumSet<DisconnectReason> BAN_REASONS =
-        EnumSet.of(
-            DisconnectReason.IRRELEVANT_NETWORK,
-            DisconnectReason.UNABLE_TO_VERIFY_NETWORK,
-            DisconnectReason.REMOTE_FAULT);
-
-    private volatile Optional<UInt64> suitableAfter = Optional.empty();
-    private final AtomicInteger score = new AtomicInteger(DEFAULT_SCORE);
-
-    public void reportInitiatedConnectionFailed(final UInt64 failureTime) {
-      suitableAfter = Optional.of(failureTime.plus(COOLDOWN_PERIOD));
-    }
-
-    public boolean shouldInitiateConnection(final UInt64 currentTime) {
-      return isSuitableAt(currentTime);
-    }
-
-    private boolean isSuitableAt(final UInt64 someTime) {
-      return suitableAfter.map(suitableAfter -> suitableAfter.compareTo(someTime) < 0).orElse(true);
-    }
-
-    public void reportInitiatedConnectionSuccessful() {
-      suitableAfter = Optional.empty();
-    }
-
-    public void reportDisconnection(
-        final UInt64 disconnectTime,
-        final Optional<DisconnectReason> reason,
-        final boolean locallyInitiated) {
-      if (isLocallyConsideredUnsuitable(reason, locallyInitiated)
-          || reason.map(DisconnectReason::isPermanent).orElse(false)) {
-        suitableAfter = Optional.of(disconnectTime.plus(BAN_PERIOD));
-        score.set(DEFAULT_SCORE);
-      } else if (suitableAfter.isEmpty()) {
-        suitableAfter = Optional.of(disconnectTime.plus(COOLDOWN_PERIOD));
-      }
-    }
-
-    private boolean isLocallyConsideredUnsuitable(
-        final Optional<DisconnectReason> reason, final boolean locallyInitiated) {
-      return locallyInitiated && reason.map(BAN_REASONS::contains).orElse(false);
-    }
-
-    public boolean adjustReputation(final ReputationAdjustment effect, final UInt64 currentTime) {
-      // No extra penalizing if already not suitable
-      if (!isSuitableAt(currentTime)) {
-        return score.get() <= DISCONNECT_THRESHOLD;
-      }
-      final int newScore =
-          score.updateAndGet(
-              current -> Math.min(MAX_REPUTATION_SCORE, current + effect.getScoreChange()));
-      final boolean shouldDisconnect = newScore <= DISCONNECT_THRESHOLD;
-      if (shouldDisconnect) {
-        // Prevent our node from connecting out to the remote peer for a long time
-        suitableAfter = Optional.of(currentTime.plus(BAN_PERIOD));
-        score.set(DEFAULT_SCORE);
-      }
-      return shouldDisconnect;
-    }
+      final boolean locallyInitiated);
 
-    @Override
-    public String toString() {
-      return MoreObjects.toStringHelper(this)
-          .add("suitableAfter", suitableAfter)
-          .add("score", score)
-          .toString();
-    }
-  }
+  boolean adjustReputation(final PeerAddress peerAddress, final ReputationAdjustment effect);
```

Comment:
this is a little confusing, we might want a comment on the interface to describe the what the boolean means, because the name isn't indicative of it...

it looks like the original returns `shouldDisconnect`, and our noop returns true, for example... but if the output is whether it should disconnect, our noop would be expected to return false i think...